### DISCUSSION TITLE
ResourceLoader return "" in code

### DISCUSSION THREAD

**Discussion by DJDQfff** [unoplatform/uno#15220] [Q&A]
* XAML
 
       <StackPanel >
          <TextBlock x:Uid="Text1" />
          <TextBlock Loaded="TextBlock_Loaded" />
        </StackPanel>

* Code

          private void TextBlock_Loaded (object sender , RoutedEventArgs e)
          {
              if (sender is TextBlock block)
              {
                  block.Text = Windows.ApplicationModel.Resources.ResourceLoader
                      .GetForViewIndependentUse()
                      .GetString("Text2");
              }
          }

* Resw

    ![image](https://github.com/unoplatform/uno/assets/31266256/0ef72ddc-80ae-496c-8b89-79db09de94c4)

There are two TextBlock in UI, the first use x:Uid,the second use code to load string .
The  first one can work , tut the function TextBlock_Loaded return an empty string

**Comment by jeromelaban** [comment]
Thanks for the report. Resources defined in a library, which is the case in the default uno template, require the use of a named resource loader to get access to its contents. Use the name of the library to get it.
